### PR TITLE
Accept flavor name as param

### PR DESCRIPTION
**Pull request by hkumarmk**
This patch will allow using flavor name for tempest tests rather than flavor
ids.

### CODE REVIEW

File: lib/puppet/type/tempest_config.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def should_to_s( newvalue )
           This is required as sometimes tempest need id(uuid) rather than name such as for image_ref, public_network_id.
           The values glance_image and network are valid.'
     validate do |value|
-      if value !~ /(glance_image|network)/
+      if value !~ /(glance_image|network|flavor)/
         raise(Puppet::Error, "Invalid values, Valid values are: glance_image, network")
```
Comment (sorenh) [inline lib/puppet/type/tempest_config.rb]: There's a new valid value, but the error message doesn't list it.

File: manifests/init.pp [hunk 2]
```diff
@@ -203,6 +203,38 @@
     }
   }
 
+  if ! $flavor_ref  and $flavor_name {
```
Comment (bodepd) [inline manifests/init.pp]: this conditional logic can be made a little simpler:

* put the failure cases first
* then you can just say if $x (and assume there is no case where they are both set to simplify the logic)

File: manifests/init.pp [hunk 3]
```diff
@@ -203,6 +203,38 @@
...
+    $flavor_uuid_set = 'flavor'
+    $flavor_ref_orig = $flavor_name
+  } elsif ! $flavor_name and $flavor_ref {
+    $flavor_ref_orig = $flavor_ref
+  } elsif ($flavor_name and $flavor_ref) or (! $flavor_name and ! $flavor_ref) {
+    fail('A value for either flavor_name or flavor_ref must be provided.')
```
Comment (bodepd) [inline manifests/init.pp]: this error message does not map all of the logic that can cause it

File: manifests/init.pp [hunk 4]
```diff
@@ -203,6 +203,38 @@
...
+    $flavor_ref_alt_orig = $flavor_name_alt
+  } elsif ! $flavor_name_alt and $flavor_ref_alt {
+    $flavor_ref_alt_orig = $flavor_ref_alt
+  } elsif $flavor_name_alt and $flavor_ref_alt  {
+    fail('either flavor_name_alt or flavor_ref_alt must be provided.')
+  }
+
+  if flavor_ref_orig {
+   tempest_config {'compute/flavor_ref':
+    value  => $flavor_ref_orig,
```
Comment (bodepd) [inline manifests/init.pp]: indentation issue.

File: manifests/init.pp [hunk 5]
```diff
@@ -203,6 +203,38 @@
...
+  if ! $flavor_ref  and $flavor_name {
+    $flavor_uuid_set = 'flavor'
+    $flavor_ref_orig = $flavor_name
+  } elsif ! $flavor_name and $flavor_ref {
+    $flavor_ref_orig = $flavor_ref
+  } elsif ($flavor_name and $flavor_ref) or (! $flavor_name and ! $flavor_ref) {
+    fail('A value for either flavor_name or flavor_ref must be provided.')
+  }
+
+  if ! $flavor_ref_alt  and $flavor_name_alt {
```
Comment (bodepd) [inline manifests/init.pp]: same comments here as above.

File: lib/puppet/type/tempest_config.rb (hunk 1 above)
Comment (hkumarmk) [inline lib/puppet/type/tempest_config.rb]: this is done

File: manifests/init.pp (hunk 2 above)
Comment (hkumarmk) [inline manifests/init.pp]: this is done

File: manifests/init.pp (hunk 3 above)
Comment (hkumarmk) [inline manifests/init.pp]: done

File: manifests/init.pp (hunk 4 above)
Comment (hkumarmk) [inline manifests/init.pp]: fixed it

File: manifests/init.pp (hunk 5 above)
Comment (hkumarmk) [inline manifests/init.pp]: this is done.

**Comment by hkumarmk** [comment]
@bodepd Fixed the issues you raised. Please review it.

**Pull request merged**